test/e2e/foursquare.e2e.test.js

```diff
@@ -0,0 +1,116 @@
+const { test, expect } = require('@playwright/test');
+
+test.describe('Foursquare Places API Integration', () => {
+  let sharedPage;
+
+  test.beforeAll(async ({ browser }) => {
+    sharedPage = await browser.newPage();
+    await sharedPage.goto('/api/foursquare');
+    await sharedPage.waitForLoadState('networkidle');
+  });
+
+  test.afterAll(async () => {
+    if (sharedPage) await sharedPage.close();
+  });
+
+  test('should load page and show primary UI elements', async () => {
```

Comment:
Let's remove this, so the focus would be content that map to the APIs and not the static page elements.